integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -450,17 +465,31 @@ def test_signum(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : unary_op_df(spark, data_gen).selectExpr('signum(a)'))
 
-@pytest.mark.parametrize('data_gen', _arith_data_gens, ids=idfn)
+@pytest.mark.parametrize('data_gen', numeric_gens + _arith_decimal_gens_low_precision, ids=idfn)
 def test_unary_minus(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : unary_op_df(spark, data_gen).selectExpr('-a'))
 
-@pytest.mark.parametrize('data_gen', _no_overflow_multiply_gens + [float_gen, double_gen] + _arith_decimal_gens, ids=idfn)
+@pytest.mark.parametrize('data_gen', _arith_decimal_gens_high_precision, ids=idfn)
+@pytest.mark.skipif(is_scala213(), reason="Apache Spark built with Scala 2.13 produces inconsistent results at high precision")
```

Comment:
This is really scary. Do you have more details on how they are different? This is just a unary minus so the only check is really about overflow. At a minimum we need a follow on issue to understand this and possibly file bugs against scala/spark.

Reply:
In some cases, it throws a java.math.ArithmeticException, which seems to be the "right"(?) behavior (I guess if you negate DECIMAL_MIN, I guess the result value could overflow). But for some values, it looks like it loses precision.  This only happens in Scala 2.13.

Reply:
That is an incompatibility and we at least need a follow on issue to try and match it. We might also want to file an issue with Spark and see if they are aware of this/expect it.  My guess is that they will not care, but in any case we need to try and match it if we can.